pkg/networkservice/core/trace/common.go

```diff
@@ -20,47 +20,106 @@ package trace
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"reflect"
 	"strconv"
 
-	"github.com/networkservicemesh/sdk/pkg/tools/log"
-
+	"github.com/pkg/errors"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/reflect/protoreflect"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
 )
 
-func logRequest(ctx context.Context, request proto.Message) {
+func logRequest(ctx context.Context, request proto.Message, isClose bool) {
+	msg := "request"
+	diffMsg := "request-diff"
+	if isClose {
+		msg = "close"
+		diffMsg = "close-diff"
+	}
+
 	connInfo, ok := trace(ctx)
 	if ok && !proto.Equal(connInfo.Request, request) {
 		if connInfo.Request != nil && connInfo.Request.ProtoReflect().Descriptor().FullName() == request.ProtoReflect().Descriptor().FullName() {
 			requestDiff, hadChanges := Diff(connInfo.Request.ProtoReflect(), request.ProtoReflect())
 			if hadChanges {
-				log.FromContext(ctx).Object("request-diff", requestDiff)
+				logObjectTrace(ctx, diffMsg, requestDiff)
 			}
 		} else {
-			log.FromContext(ctx).Object("request", request)
+			logObjectTrace(ctx, msg, request)
 		}
 		connInfo.Request = proto.Clone(request)
 	}
 }
 
-func logResponse(ctx context.Context, response proto.Message) {
+func logResponse(ctx context.Context, response proto.Message, isClose bool) {
```

Comment:
Please do not use bolean logic. Add prefixes or split method into two separate.
```suggestion
func logResponse(ctx context.Context, response proto.Message, prefixes...string) {
```